src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -2340,14 +2486,23 @@ def get_test_info(test_class: MatterBaseTest, matter_test_config: MatterTestConf
 
     info = []
     for t in tests:
-        info.append(TestInfo(t, steps=base.get_test_steps(t), desc=base.get_test_desc(t), pics=base.get_test_pics(t)))
+        info.append(TestInfo(t, steps=base.get_test_steps(t), desc=base.get_test_desc(t),
+                    pics=base.get_test_pics(t), pixits=base.get_test_pixits(t)))
 
     return info
 
 
 def run_tests_no_exit(test_class: MatterBaseTest, matter_test_config: MatterTestConfig, hooks: TestRunnerHooks, default_controller=None, external_stack=None) -> bool:
 
-    get_test_info(test_class, matter_test_config)
+    # Get test info which now includes PIXITs
+    test_info = get_test_info(test_class, matter_test_config)
+
+    # Validate PIXITs before proceeding
+    validator = PIXITValidator()
+    valid, error = validator.validate_pixits(test_info[0].pixits, matter_test_config.global_test_params)
```

Comment:
Does this validate only for the first test in the list?

might be better in the setup_test function for each test.

Reply:
Yeah, thank you for noticing this issue, I've moved validation to setup_test